src/util/miscmath.c

```diff
@@ -495,3 +509,22 @@ uint64_t umuldiv64(uint64_t x, uint64_t multiplier, uint64_t divisor) {
 uint64_t uceildiv64(uint64_t x, uint64_t y) {
 	return x / y + (x % y != 0);
 }
+
+int popcnt32(uint32_t x) {
+	#if TAISEI_BUILDCONF_HAVE_BUILTIN_POPCOUNT
```

Comment:
There seems to be an inconsistency here between how this is defined by meson (without a value) in build_config.h, and by src/util/compat.h, which can lead to build failures (e.g. https://travis-ci.com/github/jon-turney/meson-corpus-test/jobs/384217603#L4514)

Reply:
Good catch, thanks; should be fixed in 0e25c3a06be326b9b78238b20145003c9ee443da. But I do have to wonder how you managed to trigger this condition in the first place - does the compiler really not have `__builtin_popcount`, or is that a Meson bug?

Reply:
The compiler I tested with does have  __builtin_popcount.

I think this issue here is that the behaviour of meson when using the `set()` method of a configuration data object with a boolean value is not what you are expecting.

https://mesonbuild.com/Configuration.html#configuring-without-an-input-file

Perhaps you want to use `set10` here instead, depending on what you've decided the meaning of this define is?

https://mesonbuild.com/Reference-manual.html#configuration-data-object

Reply:
I'm aware of how the `set` method works, that's what I want. I just accidentally wrote `#if` in place of `#ifdef`.

Disregard my previous comment by the way; I just realized that the error happens when __builtin_popcount *is* found, since the macro just expands into an empty space. It's strange that neither GCC (10.2.1) nor clang (10.0.1) produce this error on my system however, otherwise I definitely should've caught this...

Reply:
FTR, this is a Ubuntu 20.04 (focal) VM, and the compiler is `gcc (Ubuntu 9.3.0-3ubuntu1) 9.3.0`.